crates/shared/src/price_estimation/native_price_cache.rs

```diff
@@ -317,26 +322,31 @@ impl CachingNativePriceEstimator {
         tokens: &'a [H160],
         timeout: Duration,
     ) -> HashMap<H160, NativePriceEstimateResult> {
+        let mut prices = self.get_cached_prices(tokens);
         if timeout.is_zero() {
-            return self.get_cached_prices(tokens);
+            return prices;
         }
 
-        let mut collected_prices = HashMap::new();
+        let uncached_tokens: Vec<_> = tokens
+            .iter()
+            .filter(|t| !prices.contains_key(t))
+            .copied()
+            .collect();
         let price_stream = self
             .0
-            .estimate_prices_and_update_cache(tokens, self.0.max_age);
+            .estimate_prices_and_update_cache(&uncached_tokens, self.0.max_age);
```

Comment:
It looks like that would also change the sorting inside the cache since now it won't update the `requested_at` anymore for the items already in the cache.
https://github.com/cowprotocol/services/blob/cb0a061b3fb47d89212da4451424e8d55a7064dc/crates/shared/src/price_estimation/native_price_cache.rs#L152-L172

Reply:
`Inner::get_cached_price()` also updates the `request_at` timestamp [here](https://github.com/cowprotocol/services/blob/main/crates/shared/src/price_estimation/native_price_cache.rs#L86).
And even if it didn't it would still be okay since this is exactly what we did until very recently.